src/main/java/com/codeborne/selenide/SelenideTargetLocator.java

```diff
@@ -41,12 +41,57 @@ public SelenideTargetLocator(Driver driver) {
     this.delegate = webDriver.switchTo();
   }
 
+  /**
+   * Switch to frame by index
+   * NB! Order of framew can be different in different browsers, see Selenide tests.
+   *
+   * @param index index of frame (0-based)
+   */
   @Override
   @Nonnull
   public WebDriver frame(int index) {
+    return frame(Wait(), index);
+  }
+
+  /**
+   * Switch to frame by index with a configurable timeout
+   * NB! Order of frames can be different in different browsers, see Selenide tests.
+   *
+   * @param index    index of frame (0-based)
+   * @param duration the timeout duration. It overrides default Config.timeout()
+   */
+  @Nonnull
+  public WebDriver frame(int index, Duration duration) {
```

Comment:
rename parameter `duration` to `timeout`.

Reply:
renamed, thx